watchtower/wtclient/backup_task.go

```diff
@@ -0,0 +1,344 @@
+			// this session, likely because either the sweep or
+			// reward output resulted in dust.  This task will be
+			// abandoned since it cannot satisfy the policy required
+			// by the tower, or is otherwise uneconomical to backup.
+			return err
+		}
+
+		// Cache the computed output values in the backup task in
+		// advance of signing the final transaction to avoid recomputing
+		// them. We also can set the reward address to use for the
+		// tower, as this is tied to the session this task was assigned.
+		t.sweepAmt = int64(sweepAmt)
+		t.rewardAmt = int64(rewardAmt)
+		t.rewardAddress = session.RewardAddress
+
+	} else {
+		// Otherwise, this is an altruist tower that doesn't need a
+		// reward output. Derive the output value for the lone output on
+		// the justice transaction.
+		sweepAmt, err := policy.ComputeAltruistOutput(
+			t.totalAmt, int64(weightEstimate.Weight()),
+		)
+		if err != nil {
+			return err
+		}
+
+		// Cache the computed output values in the backup task in
+		// advance of signing the final transaction to avoid
+		// recomputing.
+		t.sweepAmt = int64(sweepAmt)
+	}
+
+	t.blobType = policy.BlobType
+
+	return nil
+}
+
+// craftSessionPayload is the final stage for a backupTask, and generates the
+// encrypted payload and breach hint that should be sent to the tower. This
+// method computes the final justice transaction using the bound
+// session-dependent variables, and signs the resulting transaction. The
+// required pieces from signatures, witness scripts, etc are then packaged into
+// a JusticeKit and encrypted using the breach transaction's key.
+func (t *backupTask) craftSessionPayload(sweepAddress []byte,
+	signer input.Signer) (wtdb.BreachHint, []byte, error) {
+
+	var hint wtdb.BreachHint
+
+	// First, copy over the sweep address, the pubkeys used to derive the
+	// to-local script, and the remote CSV delay.
+	keyRing := t.breachInfo.KeyRing
+	justiceKit := &blob.JusticeKit{
+		SweepAddress:     sweepAddress,
+		RevocationPubKey: toBlobPubKey(keyRing.RevocationKey),
+		LocalDelayPubKey: toBlobPubKey(keyRing.DelayKey),
+		CSVDelay:         t.breachInfo.RemoteDelay,
+	}
+
+	// If this commitment has an output that pays to us, copy the to-remote
+	// pubkey into the justice kit. This serves as the indicator to the
+	// tower that we expect the breaching transaction to have a non-dust
+	// output to spend from.
+	if t.breachInfo.LocalOutputSignDesc != nil {
+		justiceKit.CommitToRemotePubKey = toBlobPubKey(
+			keyRing.NoDelayKey,
+		)
+	}
+
+	// Now, begin construction of the justice transaction. We'll start with
+	// a version 2 transaction.
+	justiceTxn := wire.NewMsgTx(2)
+
+	// Next, add the non-dust inputs that were derived from the breach
+	// information. This will either be contain both the to-local and
+	// to-remote outputs, or only be the to-local output.
+	inputs := t.inputs()
+	for prevOutPoint := range inputs {
+		justiceTxn.AddTxIn(&wire.TxIn{
```

Comment:
We should modify the sequence numbers so they aren't all final so we can later extend the tower (and also eventually `lnd`) to be able to broadcast multiple versions of the transaction with varying fee rates a la #1221